### PR TITLE
revise(ci): update node version

### PR DESCRIPTION
**Pull request by maikelordaz**
The deprecated methods can be checked [here](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/) for the node version related and [here](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) for the `set-output` command. @elallamTaka please approve the PR just to check if it is fetching the correct branch to commit the ABIs now

### CODE REVIEW

**Review by elallamTaka — APPROVED** [review]

File: .github/workflows/02-sc_testnet_deploy_cd.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
             echo "::set-output name=branch::${{ github.head_ref }}"
```
Comment (elallamTaka) [inline .github/workflows/02-sc_testnet_deploy_cd.yml]: forgot to revise set-output here

**Comment by maikelordaz** [comment]
@elallamTaka commit 75f1232 worked to commit the ABIs on push. I'm changing it again to work only on approval reviews, please check again

**Comment by maikelordaz** [comment]
Done. Should work now @elallamTaka , it worked fine on commit ebaa45e and 75f1232, for that ones I changed it to work when push. It is set now to when a review is approved. Here is the successful action https://github.com/TakafulDAO/takaturn/actions/runs/8089493923

**Review by elallamTaka — CHANGES_REQUESTED** [review]

File: .github/workflows/02-sc_testnet_deploy_cd.yml (hunk 1 above)
Comment (maikelordaz) [reply .github/workflows/02-sc_testnet_deploy_cd.yml]: Commit b03b480

**Review by elallamTaka — APPROVED** [review]

**Review by elallamTaka — APPROVED** [review]

**Review by elallamTaka — APPROVED** [review]

**Review by elallamTaka — APPROVED** [review]

**Review by elallamTaka — APPROVED** [review]

**Review by elallamTaka — APPROVED** [review]

**Comment by maikelordaz** [comment]
rerun prints

**Comment by maikelordaz** [comment]
rerun prints

**Review by elallamTaka — APPROVED** [review]

**Pull request merged**